### PR TITLE
`print_versions` outputs more, updated GH issue template

### PR DESCRIPTION
**Pull request by flying-sheep**
See #978 and #1341

### CODE REVIEW

**Comment by ivirshup** [comment]
I was thinking we could go a bit further.

We could add `sinfo` as a dependency and make `print_versions` just call: `sinfo.sinfo(dependencies=True)` which will always be comprehensive.

<details>
<summary> Example output: </summary>


```
-----
IPython     7.16.1
scanpy      1.5.2.dev38+g6728bdab
sinfo       0.3.1
-----
IPython             7.16.1
PIL                 7.2.0
anndata             0.7.5.dev0+g58886f0.d20200729
asciitree           NA
backcall            0.2.0
cffi                1.14.0
cloudpickle         1.5.0
colorama            0.4.3
cycler              0.10.0
cython_runtime      NA
dask                2.21.0
dateutil            2.8.1
decorator           4.4.2
fasteners           NA
get_version         2.1
google              NA
h5py                2.10.0
igraph              0.8.2
ipython_genutils    0.2.0
jedi                0.17.2
joblib              0.16.0
kiwisolver          1.2.0
legacy_api_wrap     1.2
leidenalg           0.8.1
llvmlite            0.33.0
louvain             0.7.0
matplotlib          3.3.0
monotonic           NA
mpl_toolkits        NA
msgpack             1.0.0
natsort             7.0.1
numba               0.50.1
numcodecs           0.6.4
numexpr             2.7.1
numpy               1.19.0
packaging           20.4
pandas              1.0.5
parso               0.7.0
pexpect             4.8.0
pickleshare         0.7.5
pkg_resources       NA
prompt_toolkit      3.0.5
psutil              5.7.0
ptyprocess          0.6.0
pygments            2.6.1
pyparsing           2.4.7
pytz                2020.1
scanpy              1.5.2.dev38+g6728bdab
scipy               1.5.1
sinfo               0.3.1
sitecustomize       NA
six                 1.15.0
sklearn             0.23.1
sphinxcontrib       NA
storemagic          NA
tables              3.6.1
tblib               1.6.0
texttable           1.6.2
tlz                 0.10.0
toolz               0.10.0
traitlets           4.3.3
typing_extensions   NA
wcwidth             0.2.5
yaml                5.3.1
zarr                2.4.0
-----
Python 3.8.5 (default, Jul 23 2020, 15:50:11) [Clang 11.0.3 (clang-1103.0.32.62)]
macOS-10.15.6-x86_64-i386-64bit
16 logical CPU cores, i386
-----
Session information updated at 2020-07-30 19:28
```

</details>

**Comment by flying-sheep** [comment]
That would be cool. If we add a lot of text though, we need to improve the issue template too, people just don’t understand where to put their text.

**Comment by ivirshup** [comment]
How would you change it? I'd probably just wrap the section that says "paste here" with a `<details>` tag.

**Comment by flying-sheep** [comment]
Yeah, but people often manage to delete the ```` ``` ```` and so on. Maybe we can write it in a way even people who don’t know markdown know what exactly to replace. Maybe some text like “Please replace this text (and only the text) with your (traceback|code|version output)”

**Comment by ivirshup** [comment]
I guess I'm not too worried about the formatting being off, since that's easy to fix. It's so much harder to guess software versions.

**Comment by flying-sheep** [comment]
Haha of course. I just thought it might be good to tackle both, which I’ll try here

**Comment by ivirshup** [comment]
People kinda seem to do this for pandas. Could base it off their issue template?

**Comment by flying-sheep** [comment]
Their guide is also applicable to us, even if it uses a lot of pandas examples.

File: scanpy/logging.py [hunk 1]
```diff
@@ -145,11 +130,12 @@ def print_versions():
     Matplotlib and Seaborn are excluded from this.
```
Comment (ivirshup) [inline scanpy/logging.py]: The doctoring should probably be updated to something like: print versions of imported packages

File: scanpy/logging.py [hunk 2]
```diff
@@ -145,11 +130,12 @@ def print_versions():
...
     """
     from ._settings import settings
-    modules = ['scanpy'] + _DEPENDENCIES_NUMERICS
-    print(' '.join(
-        f'{mod}=={ver}'
-        for mod, ver in _versions_dependencies(modules)
-    ), file=settings.logfile)
+    stdout = sys.stdout
+    try:
+        sys.stdout = settings.logfile
```
Comment (ivirshup) [inline scanpy/logging.py]: Should this be going to the log file? The function is called `print`, and I think it's up to the logging framework what do about `stdout`.

**Review by ivirshup — COMMENTED** [review]
Minor feedback on docs and logging.

* Could we get code check to skip the .github files?

**Comment by flying-sheep** [comment]
I just disabled `RemarkLint` in codacy. Markdown is only used for GitHub stuff.

File: scanpy/logging.py (hunk 1 above)
Comment (flying-sheep) [reply scanpy/logging.py]: thank you, missed that one!

File: scanpy/logging.py (hunk 2 above)
Comment (flying-sheep) [reply scanpy/logging.py]: True, those are called by the user manually so they don’t pollute stdout anyway even if scanpy is used as a library.

**Pull request merged**